### ISSUE TITLE
moveCalendarsBy in inline calendar returns a wrong month array

### ISSUE THREAD

**Issue by koenz** [vlio20/angular-datepicker#166]
HI,

I'm using the dp-day-calendar in a project. I can do almost all necessary things with it to create a (sort-of) range datepicker, real nice. I'm looking for a functionality to jump to a certain date (end date or start date). I use the moveCalendarsBy api method but it is not working as expected.

I use the config: allowMultiSelect: true

the issue I have: It jumps to the right month, but when I click to go to other months it seems like the selected date (this.selected) is selected in every single month. I think maybe generateMonthArray includes the selected value in every month. 

Now I know the moveCalendarsBy is not described so it might be WIP. But it would be nice to have a functionality for jumping to a specific month. Especially for multiple select. 

Thanks for your work!

**Comment by vlio20** [comment]
Hi @koenz, so if I will create an API which a capability of moving the calendars to a specific date, will it solve you issue?

Also, could you please post a link of you range selctor implementation?

**Comment by koenz** [comment]
Hey, thanks for your fast reply. This would definitely help me!

The project is protected but I can definitely send the range picker implementation. I'm trying to recreate the ww.airbnb.com implementation of the datepicker (only with a single datepicker). therefor I will have refactor that a little bit. Right now it is just a bunch of if statements.

In order to complete my range picker implementation I found out I need one more thing. Namely, some classes to style the whole thing. I made a small styling draft (https://codepen.io/anon/pen/jLmyZr) but I found out it didn't work because the weekdays are wrapped. This would probably solve it:

1. Add a class to the selected to identify the selected: .is-selected-index-1, .is-selected-index-2
2. Add class to the month to identify previous selected: .has-previous-selected
3. Add classes to weeks to identify is in range (Or un-wrap): .is-in-range
Or maybe there is a better solution but I haven't thought of it
